crypto/asn1/asn1_parse.c

```diff
@@ -43,7 +44,8 @@ static int asn1_print_info(BIO *bp, long offset, int depth, int hl, long len,
     }
     if (bp != NULL) {
         if (BIO_set_prefix(bp, str) <= 0) {
-            if ((bp = BIO_push(BIO_new(BIO_f_prefix()), bp)) == NULL)
+            if ((bio = BIO_new(BIO_f_prefix())) == NULL
```

Comment:
Note that BIO_push(b1, b2) does if (b1 == NULL) return b2

Reply:
Exactly.  The unconditional free later on then free's _bp_ and destroys it for the caller which is wrong.  The free has to happen only if the new BIO was successfully added to the chain.